### ISSUE TITLE
Feature request: castext-proof version of stack_disp_comma_separate()

### ISSUE THREAD

**Issue by mkraska** [maths/moodle-qtype_stack#917]
If you define a variable 

    t5: castext("{@stack_disp_comma_separate([a,b,c_12])@}");

and display it using

    <p>{@t5@} \({@t5@}\)</p>

you get

![image](https://user-images.githubusercontent.com/12987918/213864581-eff4ce14-4cfa-4a0f-954a-3fd003f28536.png)

This is because the function returns a string without applying TeX stuff. 

I'd like to see from `<p>{@t5@}</p>`

![image](https://user-images.githubusercontent.com/12987918/213865010-9f4a6703-69de-4f77-8cd7-fd0c73ca75a9.png) from `<p>{@t5@}</p>`

**Comment by aharjula** [comment]
Something like this?
```
stack_disp_comma_separate_math_castext(_list_of_things):= castext("[[define _first='true'/]][[foreach _thing='_list_of_things']][[if test='not _first']], [[else]][[define _first='false'/]][[/if]]{@_thing@}[[/foreach]]");
```

This is again one of those places where it is almost simpler to write the logic as CASText and compile that to normal Maxima if one wants to include it to core libraries. Although for this particular use case, a proper `lreduce` would probably be better:
```
ct_c_s_md(_L):= if length(_L) = 0 then "" else if length(_L) = 1 then castext("{@_L[1]@}",md) else (
 lreduce(castext_concat, rest(join(makelist(castext("{@_Li@}",md),_Li,_L), makelist(", ",length(_L))),-1))
);
```
Take a wild guess on what that `md` does there and what happens if you remove it. The original example would similarly behave differently if that `md` argument gets added.

**Comment by mkraska** [comment]
Yes, actually the [[ ]]-based version is already implemented in a Meclib helper function (based on your recommendation in spring 2022). I just would recommend to shift this to the core. 

- with `md`: result is good for display in Markdown environment, 
- without `md` it is good for display in html environment 

I usually produce feedback text for html, because the yellow background won't work if PRT text format is set to Markdown.

```
{@ct_c_s_md([a,b,c_12])@}

<p>{@ct_c_s([a,b,c_12])@}</p>
```

Both versions produce $a,b,c_{12}$ in a Markdown castext region.
